core/types/gen_header_json.go

```diff
@@ -12,24 +12,25 @@ import (
 )
 
 func (h Header) MarshalJSON() ([]byte, error) {
-	type HeaderJSON struct {
-		ParentHash  common.Hash    `json:"parentHash"`
-		UncleHash   common.Hash    `json:"sha3Uncles"`
-		Coinbase    common.Address `json:"miner"`
-		Root        common.Hash    `json:"stateRoot"`
-		TxHash      common.Hash    `json:"transactionsRoot"`
-		ReceiptHash common.Hash    `json:"receiptsRoot"`
-		Bloom       Bloom          `json:"logsBloom"`
-		Difficulty  *hexutil.Big   `json:"difficulty"`
-		Number      *hexutil.Big   `json:"number"`
-		GasLimit    *hexutil.Big   `json:"gasLimit"`
-		GasUsed     *hexutil.Big   `json:"gasUsed"`
-		Time        *hexutil.Big   `json:"timestamp"`
-		Extra       hexutil.Bytes  `json:"extraData"`
-		MixDigest   common.Hash    `json:"mixHash"`
-		Nonce       BlockNonce     `json:"nonce"`
-	}
-	var enc HeaderJSON
+	type Header struct {
+		ParentHash  common.Hash    `json:"parentHash"       gencodec:"required"`
+		UncleHash   common.Hash    `json:"sha3Uncles"       gencodec:"required"`
+		Coinbase    common.Address `json:"miner"            gencodec:"required"`
+		Root        common.Hash    `json:"stateRoot"        gencodec:"required"`
+		TxHash      common.Hash    `json:"transactionsRoot" gencodec:"required"`
+		ReceiptHash common.Hash    `json:"receiptsRoot"     gencodec:"required"`
+		Bloom       Bloom          `json:"logsBloom"        gencodec:"required"`
+		Difficulty  *hexutil.Big   `json:"difficulty"       gencodec:"required"`
+		Number      *hexutil.Big   `json:"number"           gencodec:"required"`
+		GasLimit    *hexutil.Big   `json:"gasLimit"         gencodec:"required"`
+		GasUsed     *hexutil.Big   `json:"gasUsed"          gencodec:"required"`
+		Time        *hexutil.Big   `json:"timestamp"        gencodec:"required"`
+		Extra       hexutil.Bytes  `json:"extraData"        gencodec:"required"`
+		MixDigest   common.Hash    `json:"mixHash"          gencodec:"required"`
+		Nonce       BlockNonce     `json:"nonce"            gencodec:"required"`
```

Comment:
All tag strings are copied over into the intermediate marshaling type. gencodec parses the tag using package reflect and doesn't know anything about the format. Removing the tags would mean that gencodec needs to construct a new tag with the `gencodec:"..."` part removed. Feel free to open a PR that does this if the extra tags bother you, but there is zero benefit :)